### DISCUSSION TITLE
Is Mapbox still compatible as a component for an Apache project?

### DISCUSSION THREAD

**Discussion by Aalbedon** [apache/superset#27580] [General]
I know that this was only tangentially mentioned in https://github.com/apache/superset/issues/9281, but with the addition of mapbox-gl-js v2.10.0 in 2.1.0+ and v2+ of mapbox-gl effectively becoming closed source, has any consideration been given to move away from Mapbox in Superset?
Or am I way off base with my interpretation of the Mapbox TOS as it relates to mapbox-gl and how it's used in the map plugins?

**Comment by michael-s-molina** [comment]
@rusackas @kgabryje @villebro Does any of you know the answer for this question?

**Reply by rusackas** [reply]
Sadly, I don't (yet) know all the pertinent details to have an informed opinion. I'd love to see alternatives proposed/implemented, particularly if they're pluggable. Ideally we'd support ALL the options we can, favoring an open-source-first approach. This will require some research, and a rebooted SIP.

**Reply by Aalbedon** [reply]
Thanks @rusackas. The reason I asked is because when I was trying to understand what's involved with Mapbox I discovered points 1.9, 1.12, and 2.14 in the product terms (https://www.mapbox.com/legal/product-terms) which are linked to from the mapbox-gl github (https://github.com/mapbox/mapbox-gl-js?tab=License-1-ov-file).

Deck.gl does seem to work best with Mapbox although some efforts out there are looking to create a deck.gl + Leaflet implementation (https://github.com/zakjan/deck.gl-leaflet).

**Reply by Aalbedon** [reply]
An update to this @rusackas, there also seems to be an open source fork available called MapLibre (https://maplibre.org/ and GitHub: https://github.com/maplibre/maplibre-gl-js), which might be an easier migration.

**Reply by rusackas** [reply]
I know that I, and I suspect "we" (the community) would be open to moving from MapBox to Map Libre. If it's more or less a drop-in replacement, is license-compatible (seems like it is), and provides feature parity, I'd call it a win. 

Seems like it could be added to DeckGL and/or the MapBox plugin (the latter of which might warrant renaming). Actually, the MapBox plugin is a "legacy" plugin using the old data API. It might be nice to simply deprecate that plugin in favor of a newer one that's TypeScript and has all the bells and whistles offered by Map Libre and by Superset (crossfiltering, drilling, etc).

If there's outcry to keep MapBox for some reason, it seems like we could either make this configurable for a deployment, configurable at the chart level, or simply use MapBox if and only if there's a MapBox API key provided in the config.

I'm rambling a lot here, but I like the idea... I think it would need a SIP for folks to discuss and agree on. Let me know if I can help with that process, but if you want to propose and contribute this, you have my support :)

**Reply by Aalbedon** [reply]
@rusackas I have submitted an SIP here: https://github.com/apache/superset/issues/28356
It being my first one, I hope it meets the requirements of each section! :)
